src/login/logind-button.c

```diff
@@ -275,6 +312,35 @@ static int button_dispatch(sd_event_source *s, int fd, uint32_t revents, void *u
                                 manager_handle_action(b->manager, INHIBIT_HANDLE_HIBERNATE_KEY, b->manager->handle_hibernate_key, b->manager->hibernate_key_ignore_inhibited, true);
                         }
                         break;
+
+                case KEY_ESC:
+                        if (b->manager->handle_secure_attention_key_press != HANDLE_IGNORE) {
+                                if (b->mods_depressed == (Button_Modifier_Ctrl | Button_Modifier_Alt | Button_Modifier_Shift)) {
+                                        log_debug("Secure Attention Key sequence pressed short.");
+                                        secure_attention_key_handler(b);
```

Comment:
please go through manager_handle_action() and invoke secure_attention_key_handler() from there, i.e. like the screen lock stuff is done. otherwise the inhibitor logic won#t work.